apps/studio/src/components/TabQueryEditor.vue

```diff
@@ -544,6 +544,22 @@
             }
           })
 
+          this.editor.on("paste", (cm, e) => {
+            e.preventDefault(); 
+            const quotes = ["'", '"', '`'];
+            let clipboard = this.$native.clipboard.readText();
+            const isQuoted = quotes.includes(clipboard[0]) && quotes.includes(clipboard[clipboard.length - 1]) && clipboard[0] === clipboard[clipboard.length - 1];
```

Comment:
So I think you should get the pasted text from CM, not the native clipboard, otherwise they could be different.
Questions:

- If someone is legit copying a quoted string from one editor to another, what will happen?
- As a new dev I have noooo idea what this is doing, can you add some comments to walk through why this is here and how it works.
- Can you abstract this to another file and add a unit test?

Reply:
Okay so after coming back from this, I'm not sure we should get the pasted text from CM, or even if we can. This event is fired before the paste is handled by code mirror, so the content isn't actually there for us to manipulate. 

Also, since this was implemented, we've added vim mode which is gonna need its own solution, as this works with copy/paste but not with yank/paste, as yank is handled by the internal vim api, and yanked text is not yanked to the system clipboard (even when it should be ie. when using `"+y`	 ). We can listen for vim commands and do much the same thing on paste, even possibly fix the lack of global yank in the process. 

Thoughts?

Reply:
I forgot to add answers to the questions lol:
1. copying from one codemirror to another works fine (there was a whitespace issue, but I've fixed that), as codemirror just copies/pastes to/from the system clipboard, unless in vim mode and you're using yank/put, which doesn't work yet
2. I'll document it for sure!
3. Same with tests :smile: